File: cinema-tickets-java/src/main/java/uk/gov/dwp/uc/pairtest/exception/InvalidPurchaseException.java
package uk.gov.dwp.uc.pairtest.exception;

/**
 * General exception
 */
public class InvalidPurchaseException extends RuntimeException {
  /**
   * Constructor
   * Generates an exception using a formatted string for the message
   *
   * @param format the format
   * @param args   the arguments
   */
  public InvalidPurchaseException(String format, Object... args) {
    super(String.format(format, args));
  }
}

File: src/pairtest/lib/InvalidPurchaseException.js
/**
 * Custom exception thrown when ticket purchase validation fails.
 * 
 * This exception is thrown when business rules are violated during
 * ticket purchase operations, such as invalid account IDs, ticket
 * quantity limits, or dependency rules.
 * 
 * @class InvalidPurchaseException
 * @extends {Error}
 */
export default class InvalidPurchaseException extends Error {
    /**
     * Creates a new InvalidPurchaseException instance.
     * 
     * @param {string} message - The error message describing the validation failure
     * @example
     * throw new InvalidPurchaseException("Account ID must be a positive integer");
     */
    constructor(message) {
        super(message);
        this.name = 'InvalidPurchaseException';
    }
}
